SmartDeviceLink/private/SDLMenuManager.m

```diff
@@ -552,7 +552,34 @@ - (BOOL)sdl_isDynamicMenuUpdateActive:(SDLDynamicMenuUpdatesMode)dynamicMenuUpda
 }
 
 - (BOOL)sdl_artworkNeedsUpload:(SDLArtwork *)artwork {
-    return (artwork != nil && ![self.fileManager hasUploadedFile:artwork] && !artwork.isStaticIcon);
+    if (artwork != nil) {
+        if (artwork.isStaticIcon) {
+            return NO;
+        } else {
+            return (artwork.overwrite || (self.fileManager != nil && ![self.fileManager hasUploadedFile:artwork]));
+        }
+    }
+    return NO;
+}
+
+- (BOOL) sdl_allArtworksUploaded:(NSArray<SDLMenuCell *> *)cells {
+    for (SDLMenuCell *cell in cells) {
+        if (![self sdl_artworkUploaded:cell.icon]) {
+            return NO;
+        }
+        if (cell.subCells != nil && (cell.subCells.count) > 0) {
+            return [self sdl_allArtworksUploaded:cell.subCells];
+        }
+    }
+
+    return YES;
+}
+
+- (BOOL) sdl_artworkUploaded:(SDLArtwork *)artwork {
+    if (artwork != nil) {
+        return artwork.isStaticIcon || (self.fileManager != nil && [self.fileManager hasUploadedFile:artwork]);
+    }
+    return YES;
```

Comment:
```suggestion
    return NO;
```

If the artwork is `nil`, it's not uploaded, right? Note, this may break some things, so it may need a refactor. Returning `YES` just doesn't make sense here.